pkg/installmanager/installmanager.go

```diff
@@ -687,23 +688,54 @@ func cleanupFailedProvision(dynClient client.Client, cd *hivev1.ClusterDeploymen
 			return err
 		}
 	case cd.Spec.Platform.IBMCloud != nil:
+		// Create IBMCloud Client
+		ibmCloudAPIKey := os.Getenv(constants.IBMCloudAPIKeyEnvVar)
+		if ibmCloudAPIKey == "" {
+			return fmt.Errorf("No %s env var set, cannot proceed", constants.IBMCloudAPIKeyEnvVar)
+		}
+		ibmClient, err := ibmclient.NewClient(ibmCloudAPIKey)
+		if err != nil {
+			return errors.Wrap(err, "Unable to create IBM Cloud client")
+		}
+		// Retrieve CISInstanceCRN
+		zones, err := ibmClient.GetDNSZones(context.TODO())
```

Comment:
This is duplicated in [contrib/pkg/deprovision/ibmcloud.go](https://github.com/openshift/hive/pull/1706/files#diff-9b605f3085fc9053107d1121f6b38e00c30c59e0522bc018bcd65afe8a194d96) and should be put somewhere common for deprovision and installmanager to call.

Reply:
I added funcs for both of these to the client.